File: NGUIProj/Assets/Scripts/UI/test/testModel.cs
using UnityEngine;
using System.Collections;

public class testModel : NotifyPropChanged//, INotifyPropChanged
{
    public const string COUNT = "test_count";

    private int m_count = 0;
    public int Count
    {
        get
        {
            return m_count;
        }
        set
        {
            m_count = value;
            OnPropertyChanged(COUNT, m_count);
        }
    }
}

File: NGUIProj/Assets/LuaFramework/Lua/MVC/test/testModel.lua
require "Framework/NotifyPropChanged"

testModel = class(NotifyPropChanged)
local this = {};

function testModel:ctor()
	this.m_count = 0;
	this.m_double = 0;
	self.KEY_COUNT = "TESTModel_Count" --绑定UI组件的唯一ID
	self.TESTMODEL_DOUBLE = "TESTMODEL_DOUBLE"
end

function testModel:getCount() 
	return this.m_count;
end

function testModel:setCount( count )
	this.m_count = count;
	--通知UI组件更新
	self:OnPropertyChanged(self.KEY_COUNT, this.m_count);
end

function testModel:getDouble()
	return this.m_double;
end

function testModel:setDouble( data )
	this.m_double = data
	self:OnPropertyChanged(self.TESTMODEL_DOUBLE, this.m_double)
end
